framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts

```diff
@@ -276,6 +276,39 @@ describe('MessageRecoveryCommand', () => {
 			expect(result.error?.message).toInclude(`Cross-chain message is not pending.`);
 		});
 
+		it('should return error if ccm indices exceed outbox size', async () => {
+			ccms = [
+				{
+					nonce: BigInt(0),
+					module: MODULE_NAME_INTEROPERABILITY,
+					crossChainCommand: CROSS_CHAIN_COMMAND_REGISTRATION,
+					sendingChainID: utils.intToBuffer(2, 4),
+					receivingChainID: utils.intToBuffer(3, 4),
+					fee: BigInt(1),
+					status: CCMStatusCode.FAILED_CCM,
+					params: Buffer.alloc(0),
+				},
+			];
+			ccmsEncoded = ccms.map(ccm => codec.encode(ccmSchema, ccm));
+			transactionParams.crossChainMessages = [...ccmsEncoded];
+			transactionParams.idxs = [2 ** terminatedChainOutboxSize + terminatedChainOutboxSize];
```

Comment:
This value must also be updated. You could use [getHeight](https://github.com/LiskHQ/lisk-sdk/blob/93d9291a0d8b8818fd7eb05c53520d988a26cbc3/elements/lisk-tree/src/merkle_tree/utils.ts#L48) to compute the edge case:
```
indexLength = getHeight(terminatedChainOutboxSize);
idxs = [terminatedChainOutboxSize + 2**indexLength]
```

Reply:
Updated, since the `getHeight` from merkle tree is not exposed, I copied the code to this file